common/persistence/persistenceInterface.go

```diff
@@ -394,12 +394,19 @@ type (
 
 	// InternalDeleteHistoryBranchRequest is used to remove a history branch
 	InternalDeleteHistoryBranchRequest struct {
-		// branch to be deleted
-		BranchInfo *persistencespb.HistoryBranch
 		// Used in sharded data stores to identify which shard to use
-		ShardID int32
-		// Max EndNodeID of each  branch. This is used to determine the range of nodes that
-		BranchesMaxEndNodeID map[string]int64
+		ShardID  int32
+		TreeId   string
+		BranchId string
+		// branch ranges to be deleted
+		BranchRanges []InternalDeleteHistoryBranchRange
+	}
+
+	// InternalDeleteHistoryBranchRange is used to delete a range of branch
+	InternalDeleteHistoryBranchRange struct {
+		TreeId      string
+		BranchId    string
+		BeginNodeId int64 // delete nodes with ID <= BeginNodeId
```

Comment:
tree ID and branch ID are specified in both req and branch range?

Reply:
There could be ancestors which is different BranchId. We could get rid of TreeId here as it is always the same.